login page: Show form-independent errors even if email auth disabled.

mateuszmandera opened the pull request:
These used to only be shown conditional on the
{% if password_auth_enabled %} in the template. Meaning that if you had an org with email auth disabled and a deactivated user tried to log in, they wouldn't see the error shown and get confused.

This switches the position of where these error will be shown (above the login+password form instead of below it), but it looks fine.

Here's how it looks now, with, or without, the login form:
![Screenshot from 2022-10-11 22-06-51](https://user-images.githubusercontent.com/45007152/195190382-545516a4-15d2-497c-8bcd-6d6a5607096f.png)
![Screenshot from 2022-10-11 22-05-59](https://user-images.githubusercontent.com/45007152/195190387-1f875a11-1fb8-4152-b9ab-bf8b5420b37a.png)

alya commented:
Can I get a screenshot of the current UI to compare? I don't know how to easily view it in dev.

alya commented:
Visually, I would add more spacing below the red notice in the screenshots in the PR description.

mateuszmandera commented:
> Can I get screenshots of the current UI to compare? I don't know how to easily view it in dev.

It was like this (basically the same, just below the form):
![image](https://user-images.githubusercontent.com/45007152/195432565-1653ae8d-d51b-46e0-b90d-58005f0ecd87.png)

If you want to test this in dev, you can just go to e.g. `http://localhost:9991/login/?is_deactivated=mateusz.mandera%40zulip.com`

mateuszmandera commented:
With the css tweak:
(subtle difference but looks a bit better i think)

![image](https://user-images.githubusercontent.com/45007152/195433377-817afefa-f9a2-4877-b60d-a961984cefcc.png)

alya commented:
Hm, I'm confused -- I'm still seeing the notice below the email area when I try to test on dev.
![Screen Shot 2022-10-12 at 11 21 22 PM](https://user-images.githubusercontent.com/2090066/195517602-e1f1823c-a570-42a1-8fd8-af6382c120a2.png)

alya commented:
Anyhow, it's not critical, but it's nice for all the notices to appear in a consistent location, and below the fields is probably more standard?

![Screen Shot 2022-10-12 at 11 22 46 PM](https://user-images.githubusercontent.com/2090066/195517862-9bf36122-f17f-4955-867a-3a168f20e4da.png)

mateuszmandera commented:
@alya 

>Hm, I'm confused -- I'm still seeing the notice below the email area when I try to test on dev.

Because I think you're testing the login via the login form. In that case, the error comes from the form, and thus is displayed below. It comes from this logic in the template instead of the one I moved in my commit:
```
                            {% if form.errors %}
                            <div class="alert alert-error">
                                {% for error in form.errors.values() %}
                                <div>{{ error | striptags }}</div>
                                {% endfor %}
                            </div>
                            {% endif %}
```
If you want to test the logic affected by my commit, you need to go through social auth, or just through the URL I mentioned earlier `http://localhost:9991/login/?is_deactivated=mateusz.mandera%40zulip.com` - then the error will appear at the top.

> Anyhow, it's not critical, but it's nice for all the notices to appear in a consistent location, and below the fields is probably more standard?

Generally agreed, but here are my two thoughts on this:
1. I think it actually kind of makes sense for the error to be shown at the top when it is not related to the form? Because the user got here through social auth, not through the login form. While login form errors are still displayed below the form.
2. Doing it differently would I think require adding more, split-up `{% if password_auth_enabled %}` segments in the template code, making it a lot messier. I'm not sure if it's worth it.

alya commented:
I tried first `http://localhost:9991/login/?is_deactivated=mateusz.mandera%40zulip.com`, but it just redirected me to ``http://localhost:9991` for some reason. 🤔

alya commented:
> I think it actually kind of makes sense for the error to be shown at the top when it is not related to the form? Because the user got here through social auth, not through the login form.

Mm, I see. Yeah, I guess I'm fine with merging it as-is.

alya commented:
@timabbott do you want to take a look?

timabbott commented:
Merged, thanks @mateuszmandera!
